### PR TITLE
feat: Added AutoMLForecastingTrainingJob and tests

### PR DESCRIPTION
**Pull request by thehardikv**
This adds the `AutoMLForecastingTrainingJob` subclass:

- [x]  Added `AutoMLForecastingTrainingJob`
- [x]  Added unit tests
- [x]  Tested [CUJ for Forecasting](http://doc/1jKpbfHHxQK_V5XjCGmdoslpPyxp3yowQePuR5q-O1MA)

Fixes [b/170511584](http://b/170511584) 🦕

### CODE REVIEW

File: google/cloud/aiplatform/training_jobs.py [hunk 1]
```diff
@@ -2580,6 +2582,447 @@ def _model_upload_fail_string(self) -> str:
...
+                [google.cloud.aiplatform.v1beta1.TrainingPipeline.training_task_definition].
+                For time series Datasets, all their data is exported to
+                training, to pick and choose from.
+            target_column (str):
+                Required. Name of the column that the Model is to predict values for.
+            time_column (str):
+                Required. Name of the column that identifies time order in the time series.
+            time_series_identifier_column (str):
+                Required. Name of the column that identifies the time series.
+            time_variant_past_only_columns (List[str]):
```
Comment (thehardikv) [inline google/cloud/aiplatform/training_jobs.py:2867]: Should we add some logic to auto-include the target column in the past only list if its not present?

**Comment by thehardikv** [comment]
Should we consider Forecasting to be a type of Tabular? So instead of referencing "Forecasting" as a standalone product in the SDK, we could use "TabularForecasting". WDYT?

File: google/cloud/aiplatform/training_jobs.py [hunk 2]
```diff
@@ -2675,16 +2671,31 @@ def run(
...
-                For tabular Datasets, all their data is exported to
+                For time series Datasets, all their data is exported to
                 training, to pick and choose from.
             target_column (str):
+                Required. Name of the column that the Model is to predict values for.
             time_column (str):
+                Required. Name of the column that identifies time order in the time series.
             time_series_identifier_column (str):
+                Required. Name of the column that identifies the time series.
             time_variant_past_only_columns (List[str]):
```
Comment (alm2) [inline google/cloud/aiplatform/training_jobs.py:2682]: I think we refer to these as time_dependent elsewhere

File: google/cloud/aiplatform/training_jobs.py [hunk 3]
```diff
@@ -2703,13 +2714,49 @@ def run(
...
                 equal weight of 1.
             static_columns (List[str]):
+                Optional. Column names that should be used as static columns.
+                Each column is constant within a time series.
             forecast_window_start (int):
+                Optional. The number of periods offset into the future as the start of the forecast window
+                (the window of future values to predict, relative to the present.), where each period is one
+                unit of granularity as defined by [period]. Inclusive.
             past_horizon (int):
-            export_evaluated_data_items_config: Optional[
```
Comment (alm11011) [inline google/cloud/aiplatform/training_jobs.py]: can we make this required?

File: google/cloud/aiplatform/training_jobs.py [hunk 4]
```diff
@@ -2703,13 +2714,49 @@ def run(
...
-                Dict[str, Union[str, bool]]
-            ] = None,
-            quantiles: Optional[List[float]] = None,
-            validation_options: Optional[str] = None,
+                Optional. The number of periods offset into the past to restrict past sequence, where each
+                period is one unit of granularity as defined by [period]. Default value 0 means that it lets
+                algorithm to define the value. Inclusive.
```
Comment (alm11011) [inline google/cloud/aiplatform/training_jobs.py:2726]: I would recommend: When not provided uses the default value of 0 which means the model sets each series historical window to be 0 (also known as "cold start").

File: google/cloud/aiplatform/training_jobs.py (hunk 1 above)
Comment (alm11011) [reply google/cloud/aiplatform/training_jobs.py:2867]: that might be worth doing server side, I can't think of any scenario where the target variable wouldn't be set as time dependent and known at prediction time

File: google/cloud/aiplatform/training_jobs.py [hunk 5]
```diff
@@ -2811,8 +2856,31 @@ def _run(
...
+                (e.g. population of a city in a given year, or weather on a given day).
+            time_variant_past_and_future_columns (List[str]):
+                Required. Column names that should be used as time variant past and future columns.
+                Each column contains information for the given entity (identified by the
+                [time_series_identifier_column]) that is known for the past and the future.
+            forecast_window_end: (int):
+                Required. The number of periods offset into the future as the end of the forecast window
+                (the window of future values to predict, relative to the present.), where each period is one unit
+                of granularity as defined by [period]. Inclusive.
+            period_unit (str):
```
Comment (alm11011) [inline google/cloud/aiplatform/training_jobs.py:2880]: can we list accepted values?

File: google/cloud/aiplatform/training_jobs.py (hunk 1 above)
Comment (thehardikv) [reply google/cloud/aiplatform/training_jobs.py:2867]: sg, let's do it server-side.

File: google/cloud/aiplatform/training_jobs.py (hunk 2 above)
Comment (thehardikv) [reply google/cloud/aiplatform/training_jobs.py:2682]: Sure, makes sense. But the naming here should be consistent with, [automl_time_series_forecasting_1.0.0.yaml](http://google3/google/cloud/aiplatform/publicfiles/trainingjob/definition/automl_time_series_forecasting_1.0.0.yaml), which is also public-facing. So let's make the shift to "time_dependent" as part of a more global re-naming. sg?

File: google/cloud/aiplatform/training_jobs.py (hunk 3 above)
Comment (thehardikv) [reply google/cloud/aiplatform/training_jobs.py]: ?

**Pull request merged**